overlord/snapstate/target.go

```diff
@@ -1135,6 +1138,13 @@ func validateAndInitStoreUpdates(allSnaps map[string]*SnapState, updates map[str
 			return err
 		}
 
+		snapName, _ := snap.SplitInstanceName(sn.InstanceName)
+		for _, add := range sn.AdditionalComponents {
+			if snapst.CurrentComponentState(naming.NewComponentRef(snapName, add)) != nil {
+				return snap.AlreadyInstalledComponentError{Component: add}
```

Comment:
Should we just log something instead of failing here? It would help to avoid some "if"s in agents/scripts I think.

Reply:
yes, I'm not sure it make sense to return already installed errors from a refresh? is this code only for refreshes?

Reply:
I dropped returning this error and now filter out additional components that are already installed.